### PR TITLE
Improve documentation and development tooling

### PR DESCRIPTION
**Pull request by kakkoyun**
#### Development tooling
* Add documentation generation make actions for dynamic content
* Remove obscure development scripts
* Introduce `tools.go` to track development tool versions
* Upgrade all tools to their latest version
* Add a CI step to check if everything generated properly
* Fix linter pointed issues after upgrade

#### Documentation
* Update README
* Update badges
* Remove deprecated Drone 0.8 documentation
* Add Makefile help to README

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -158,19 +165,18 @@ format: ; $(info $(M) running format )
...
 $(GOTEST_BIN): ; $(info $(M) getting gotest )
-	$(Q) GO111MODULE=off $(GOGET) -u github.com/rakyll/gotest
+	$(Q) $(GOBUILD) -o $@ github.com/rakyll/gotest
 
 $(EMBEDMD_BIN): ; $(info $(M) getting embedmd )
-	$(Q) GO111MODULE=off $(GOGET) -u github.com/campoy/embedmd
+	$(Q) $(GOBUILD) -o $@ github.com/campoy/embedmd
+
+$(LICHE_BIN): ; $(info $(M) getting liche )
+	$(Q) $(GOBUILD) -o $@ github.com/raviqqe/liche
```
Comment (aweris) [inline Makefile:179]: Another alternative for installing tools : 
```
# List of tools
GOTOOLS := $(shell cat tools.go | grep _ | awk -F'"' '{print $$2}')

.PHONY: install-tools
install-tools: ## Install tools
install-tools: vendor ; $(info $(M) installing tools)
	$(Q) $(GOINSTALL) $(GOTOOLS)
```

File: Makefile [hunk 2]
```diff
@@ -158,19 +165,18 @@ format: ; $(info $(M) running format )
...
 
 $(GOLANGCI_LINT_BIN): ; $(info $(M) getting golangci-lint )
 	$(Q) curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/$(GOLANGCI_LINT_VERSION)/install.sh \
```
Comment (aweris) [inline Makefile:182]: Maybe you should also move `golangci-lint` and `goreleaser` to `tools.go`

File: Makefile [hunk 3]
```diff
@@ -32,8 +32,7 @@ DOCKER_BUILD          := $(DOCKER) build
 DOCKER_PUSH           := $(DOCKER) push
 DOCKER_COMPOSE        := docker-compose
 
-
-DOCKER_REPO           :=  meltwater/drone-cache
+CONTAINER_REPO        :=  meltwater/drone-cache
```
Comment (aweris) [inline Makefile:35]: Also, I think to allow customization of `CONTAINER_REPO` is better
```suggestion
CONTAINER_REPO        ?=  meltwater/drone-cache
```

File: Makefile (hunk 1 above)
Comment (kakkoyun) [reply Makefile:180]: 👍

File: Makefile (hunk 3 above)
Comment (kakkoyun) [reply Makefile:36]: 👍

File: Makefile (hunk 2 above)
Comment (kakkoyun) [reply Makefile:183]: `golangci-lint` should be moved for sake of consistency. However, I don't think, I'll move `goreleaser`. I even think I should remove it from `Makefile`.

**Comment by kakkoyun** [comment]
@aweris Thanks for the review, all applied.

**Pull request merged**